lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,423 @@
+    except psycopg2.OperationalError as e:        
+        logger.critical("Unable to connect to database.  Connection timeout.")
+        logger.exception('%s', e)
+        sys.exit('Unable to load inbound_numbers table into dictionary')
+    except Exception as e:
+        logger.critical('Failed to query database: %s', e)
+        if db_connection:
+            db_connection.close()
+        sys.exit('Unable to load inbound_numbers table into dictionary')
+
+def set_aws_clients() -> None:
+    """
+    generate the pinpoint and sqs client to be able to use them throughout the execution cycle
+    """
+    global aws_pinpoint_client, aws_sqs_client
+
+    # If already set, no need to set it again
+    if aws_pinpoint_client is not None and aws_sqs_client is not None:
+        return
+    try:
+        logger.info('Setting aws_pinpoint_client...')
+        aws_pinpoint_client = boto3.client('pinpoint', region_name=AWS_REGION)
+        logger.info('aws_pinpoint_client set...')
+
+        logger.info('Setting aws_sqs_client...')
+        aws_sqs_client = boto3.client('sqs', region_name=AWS_REGION)
+        logger.info('aws_sqs_client set...')
+    except Exception as e:
+        logger.critical('Unable to set pinpoint client: %s', e)        
+
+def set_database() -> None:
+    """
+    Retrieve the database uri from SSM store    
+    """
+    global SQLALCHEMY_DATABASE_URI
+
+    try:
+        logger.info("Getting the database URI from SSM Parameter Store . . .")
+        logger.info(DATABASE_URI_PATH)
+
+        SQLALCHEMY_DATABASE_URI = read_from_ssm(DATABASE_URI_PATH)
+
+        logger.info("Retrieved DB configuration")
+    except Exception as e:
+        logger.info('Failed to configure database: %s', e)
+        sys.exit('Unable to configure database')
+
+def read_from_ssm(key: str) -> str:
+    """
+    Read parameter from SSM store.
+    """
+    try:
+        ssm_client = boto3.client('ssm')
+
+        logger.info("Generated ssm_client")
+
+        response = ssm_client.get_parameter(
+            Name=key,
+            WithDecryption=True
+        )
+
+        logger.info("received ssm parameter")
+
+        return response.get("Parameter", {}).get("Value", '')
+    except Exception as e:
+        logger.error("General Exception With Call to VeText")
+        logger.exception('%s', e)
+        return ''
+
+def init_execution_environment() -> None:
+    """
+    Collects environmental variables, sets up the logger, populates the two_way_sms_table_dict,
+    and sets up the aws pinpoint and sqs clients.
+    """
+    set_env_variables()
+    logger.info("Env vars set")
+    set_database()
+    logger.info("Database configured")
+    set_service_two_way_sms_table()
+    logger.info("Services loaded from database")
+    set_aws_clients()
+    logger.info("Pinpoint and SQS clients configured")
+
+    logger.info('Execution environment setup...')
+
+# init_execution_environment()
```

Comment:
Absolutely not.
The environment setup checks need to be done prior to invocations.

Reply:
If this is only while testing, fine. But it needs to be pulled out before being merged.